### PR TITLE
Disable scene

### PR DESCRIPTION
**Pull request by callemand**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [X] If your changes affects code, did your write the tests?
- [X] Are tests passing? (`npm test` on both front/server)
- [X] Is the linter passing? (`npm run eslint` on both front/server)
- [X] Did you run prettier? (`npm run prettier` on both front/server)
- [X] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [X] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

### Description of change
Adding disable scene  and change start button by test

Screenshot 2021-02-13 at 21 11 58
Screenshot 2021-02-13 at 21 11 52

### CODE REVIEW

**Review by atrovato — COMMENTED** [review]
Code is fine to me,
but I tried on my local env, and when I disable a scene, it is always enabled.
For more, why don't we show the "test" button when scene is disabled ? We may want to test is even if it is not enabled, no?

**Comment by callemand** [comment]
> Code is fine to me,

Thank you

> but I tried on my local env, and when I disable a scene, it is always enabled.

It's weird. is it on debug mode ?

> For more, why don't we show the "test" button when scene is disabled ? We may want to test is even if it is not enabled, no?

Ok, it's a mistake on my side. When I read the code, I had the impression that the scene had to be configured in node-schedule to be able to be forced but by rereading the code, I realized that it was not. I will make the changes accordingly

File: server/lib/scene/scene.addScene.js [hunk 1]
```diff
@@ -31,65 +31,68 @@ function addScene(sceneRaw) {
...
-            break;
-          default:
-            throw new BadParameters(`${trigger.unit} not supported`);
-        }
-        if (intervalMilliseconds > MAX_VALUE_SET_INTERVAL) {
-          throw new BadParameters(`${trigger.interval} ${trigger.unit} is too big for an interval`);
+
+  if (scene.active) {
+    // Foreach triggger, we schedule jobs for triggers that need to be scheduled
+    if (scene.triggers) {
```
Comment (atrovato) [inline server/lib/scene/scene.addScene.js]: Why not both if in single one?

If scene is active and have trigger, no.

**Review by atrovato — CHANGES_REQUESTED** [review]
A simple code comment, and build now fails.
Once both fixed, I'll try it again.

File: server/migrations/202102212184604-add-active-scene.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return Promise.all([
+      queryInterface.addColumn('t_scene', 'active', {
+        type: Sequelize.BOOLEAN,
+        defaultValue: true,
```
Comment (Pierre-Gilles) [inline server/migrations/202102212184604-add-active-scene.js:6]: we might want a `allowNull: false,` here

File: server/models/scene.js [hunk 3]
```diff
@@ -115,6 +115,10 @@ module.exports = (sequelize, DataTypes) => {
       last_executed: {
         type: DataTypes.DATE,
       },
+      active: {
+        type: DataTypes.BOOLEAN,
+        defaultValue: true,
```
Comment (Pierre-Gilles) [inline server/models/scene.js:121]: same, `allowNull: false,`

File: server/lib/scene/scene.addScene.js (hunk 1 above)
Comment (callemand) [reply server/lib/scene/scene.addScene.js]: You are right 👍

**Comment by atrovato** [comment]
@callemand 
Same issue, scenes is never disabled.

> > but I tried on my local env, and when I disable a scene, it is always enabled.

> It's weird. is it on debug mode ?

Yes it is on debug mode.
No log displayed when I switch scene on/off.
I think, you miss a "active" condition on scene.executeActions.js (at execution time).

**Comment by callemand** [comment]
@atrovato 
Thank for your return

> Same issue, scenes is never disabled.
> 
> > > but I tried on my local env, and when I disable a scene, it is always enabled.
> 
> > It's weird. is it on debug mode ?
> 
> Yes it is on debug mode.
> No log displayed when I switch scene on/off.

Normally, the scene is disable now :). I had only tested on the periodic trigger because i don't have any device connected on my dev environment.

> I think, you miss a "active" condition on scene.executeActions.js (at execution time).

Yes, I have miss an "active" condition. I have added a condition into scene.execute.js not into scene.executeAction.

**Comment by cicoub13** [comment]
Will fix #980

**Comment by atrovato** [comment]
Now you miss tests :/

**Comment by callemand** [comment]
> Now you miss tests :/

@atrovato
Sorry, but I don't understand, all unit test works.

But the pipeline failed with this error: '85.71% of diff hit (target 95.55%)'
But on the top of the PR the codecov return: 
```
@@           Coverage Diff           @@
##           master    #1075   +/-   ##
=======================================
  Coverage   95.55%   95.55%           
=======================================
  Files         531      531           
  Lines        7112     7114    +2     
=======================================
+ Hits         6796     6798    +2     
  Misses        316      316      
```     

What i missed ?

**Comment by atrovato** [comment]
What I understand from codecov comment, 

`server/lib/scene/scene.execute.js | 92.85% <83.33%> (+1.19%)` is not tested enough.

File: server/test/lib/scene/scene.execute.test.js [hunk 4]
```diff
@@ -71,4 +71,27 @@ describe('SceneManager', () => {
...
+          {
+            type: 'BAD-TYPE',
+            devices: ['light-1'],
+          },
+        ],
+      ],
+      active: true,
+    };
+    sceneManager.addScene(scene);
+    await sceneManager.execute('my-bad-scene');
```
Comment (atrovato) [inline server/test/lib/scene/scene.execute.test.js:97]: But no assertion here!

**Review by atrovato — CHANGES_REQUESTED** [review]
Test added, but no assert.

File: server/test/lib/scene/scene.execute.test.js (hunk 4 above)
Comment (callemand) [reply server/test/lib/scene/scene.execute.test.js:97]: Yes, I know, it was just to test the codecov return.

**Review by atrovato — APPROVED** [review]
Ok, approved 👍

**Comment by Pierre-Gilles** [comment]
Hey! Thanks for your PR. 

I'm not really sure this is the behavior we want.

It's weird to me that a disabled scene could not be tried...

To me, the "disable scene" feature is more a "disable trigger for this scene" feature, not a "disable the scene completely".

I think people want to disable a scene while they are still building it, or if they want to pause the automatic start of the scene (while they are in holiday for example). But still, I think that people still want to be able to do manual runs. 

What do you think about that?

**Comment by atrovato** [comment]
That's right, we already talked about the fact we need to keep the "test" button,
but since you added the test on active in `scene.execute.js`, test is disabled.
Maybe the "active" test should be only on triggers (all triggers).

**Comment by Pierre-Gilles** [comment]
Thanks for the development! 

I merged a modified version in https://github.com/GladysAssistant/Gladys/pull/1229

**Pull request closed without merging**